ui/src/views/Job.vue

```diff
@@ -62,43 +62,45 @@
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue'
-import { Job } from '@cube-creator/model'
 import type { CreativeWork } from '@rdfine/schema'
 import { schema } from '@tpluscode/rdf-ns-builders'
+import { computed, defineComponent } from 'vue'
+import { useStore } from 'vuex'
+import { useRoute } from 'vue-router'
+
 import BMessage from '@/components/BMessage.vue'
 import ExternalTerm from '@/components/ExternalTerm.vue'
-import JobStatus from '../components/JobStatus.vue'
-import LoadingBlock from '../components/LoadingBlock.vue'
-import MarkdownRender from '../components/MarkdownRender.vue'
-import ValidationReportDisplay from '../components/ValidationReportDisplay.vue'
-import { mapGetters } from 'vuex'
+import JobStatus from '@/components/JobStatus.vue'
+import LoadingBlock from '@/components/LoadingBlock.vue'
+import MarkdownRender from '@/components/MarkdownRender.vue'
+import ValidationReportDisplay from '@/components/ValidationReportDisplay.vue'
+import { RootState } from '@/store/types'
+import { usePolling } from '@/use-polling'
 
 export default defineComponent({
   name: 'JobView',
   components: { BMessage, ExternalTerm, JobStatus, LoadingBlock, MarkdownRender, ValidationReportDisplay },
 
-  computed: {
-    ...mapGetters('project', {
-      findJob: 'findJob',
-    }),
+  setup () {
+    const store = useStore<RootState>()
+    const route = useRoute()
 
-    language (): string[] {
-      return this.$store.state.app.language
-    },
+    const findJob = store.getters['project/findJob']
+    const jobId = route.params.jobId
+    const job = computed(() => findJob(jobId))
+    const link = computed(() => job.value?.link?.id.value)
+    const createdDate = computed(() => job.value?.created.toLocaleString() ?? '')
 
-    job (): Job | undefined {
-      const jobId = this.$route.params.jobId
-      return this.findJob(jobId)
-    },
+    usePolling(() => store.dispatch('project/fetchJob', jobId))
```

Comment:
What magic causes `onMounted/onUnmounted` to be called in the right moments for the right component?

Reply:
I think that should help https://vuejs.org/guide/extras/composition-api-faq.html#what-is-composition-api

In summary, Vue's "new" composition API allows for better reuse between components. It looks a bit like React hooks if you've ever used them (although it doesn't work the same).